tests/test_rest_bucket.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bats
+
+# Copyright 2024 Versity Software
+# This file is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+load ./bats-support/load
+load ./bats-assert/load
+
+source ./tests/commands/get_object_lock_configuration.sh
+source ./tests/commands/head_bucket.sh
+source ./tests/commands/list_buckets.sh
+source ./tests/logger.sh
+source ./tests/setup.sh
+source ./tests/util/util_bucket.sh
+source ./tests/util/util_list_buckets.sh
+source ./tests/util/util_lock_config.sh
+source ./tests/util/util_ownership.sh
+source ./tests/util/util_rest.sh
+source ./tests/util/util_tags.sh
+
+@test "REST - HeadBucket" {
+  run setup_bucket "s3api" "$BUCKET_ONE_NAME"
+  assert_success
+
+  run head_bucket_rest "$BUCKET_ONE_NAME"
+  assert_success
+}
+
+@test "REST - HeadBucket - doesn't exist" {
+  run head_bucket_rest "$BUCKET_ONE_NAME"
+  assert_failure 1
+}
+
+@test "REST - bucket tagging - no tags" {
+  run setup_bucket "s3api" "$BUCKET_ONE_NAME"
+  assert_success
+
+  run verify_no_bucket_tags_rest "$BUCKET_ONE_NAME"
+  assert_success
+}
+
+@test "REST - bucket tagging - tags" {
+  test_key="testKey"
```

Comment:
Would there be some value in testing a range of key/value pairs (with numbers, symbols, length)?